### PR TITLE
Fee reserve for lightning backends

### PR DESCRIPTION
**Pull request by callebtc**
Reimplements fee reserve from the commits that were lost in the FastAPI transition for `LndRestWallet` and `CLightningWallet`, originally implemented with the help of @iwarpbtc. 

Note: `LndWallet` (GRPC) also has a fee reserve implemented here but the wallet has been phased out for now. Issues that remain were the check for the status after a successful payment. 

12 File changes because `pay_invoice` of every wallet has now the (often unused) argument `fee_limit_msat`.

### CODE REVIEW

File: lnbits/core/services.py [hunk 1]
```diff
@@ -134,26 +133,16 @@ async def pay_invoice(
...
-                wallet.balance_msat / 100 * 2
-            ):
-                raise PermissionError(
-                    "LNbits requires you keep at least 2% reserve to cover potential routing fees."
-                )
+        # do the balance check
+        wallet = await get_wallet(wallet_id, conn=conn)
+        assert wallet
+        if wallet.balance_msat < 0:
+            raise PermissionError("Insufficient balance.")
```
Comment (callebtc) [inline lnbits/core/services.py:145]: Restore original balance checking, will respect fee for internal vs external payments.

File: lnbits/wallets/lndgrpc.py [hunk 2]
```diff
@@ -143,10 +144,10 @@ async def create_invoice(
...
         return InvoiceResponse(True, checking_id, payment_request, None)
 
-    async def pay_invoice(self, bolt11: str) -> PaymentResponse:
-        resp = await self.rpc.SendPayment(
-            lnrpc.SendPaymentRequest(payment_request=bolt11)
-        )
+    async def pay_invoice(self, bolt11: str, fee_limit_msat: int) -> PaymentResponse:
+        fee_limit_fixed = ln.FeeLimit(fixed=fee_limit_msat // 1000)
+        req = ln.SendRequest(payment_request=bolt11, fee_limit=fee_limit_fixed)
+        resp = await self.rpc.SendPaymentSync(req)
```
Comment (callebtc) [inline lnbits/wallets/lndgrpc.py:150]: works

**Comment by arcbtc** [comment]
Wow, well done. I notice fee_limit_msat being added to lnpay, opennode and lntxbot, but not being used?

**Comment by arcbtc** [comment]
Ah, I see, services needs it

**Review by iWarpBTC — APPROVED** [review]

**Review by arcbtc — APPROVED** [review]

**Pull request merged**